File: versioned_docs/version-1.20.x/examples/go/src/data_ingestion_batch.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	"time"
)

func toTimestamp(t string) int64 {
	timestamp, err := time.Parse(time.RFC3339, t)
	if err != nil {
		panic(err)
	}
	return timestamp.UnixMicro()
}

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Get or create a bucket with the name "my-bucket"
	bucket, err := client.CreateOrGetBucket(context.Background(), "my-bucket", nil)
	if err != nil {
		panic(err)
	}

	ctx := context.Background()

	// Prepare a batch of records for different entries
	batch := bucket.BeginWriteRecordBatch(ctx)

	batch.Add("sensor-1", toTimestamp("2024-02-02T10:00:00Z"), []byte("Temperature: 20.5"), "text/plain", reduct.LabelMap{"unit": "C"})
	batch.Add("sensor-1", toTimestamp("2024-02-02T10:00:01Z"), []byte("Temperature: 20.6"), "text/plain", reduct.LabelMap{"unit": "C"})
	batch.Add("camera-1", toTimestamp("2024-02-02T10:00:02Z"), []byte("Frame #1"), "text/plain", nil)

	// Write records from multiple entries in one request
	errMap, err := batch.Send(ctx)
	if err != nil {
		panic(err)
	}

	// Check statuses and raise first error
	if len(errMap) > 0 {
		for entry, entryErrors := range errMap {
			for timestamp, err := range entryErrors {
				panic("Error writing record to " + entry + " at timestamp " + time.UnixMicro(timestamp).Format(time.RFC3339) + ": " + err.Error())
			}
		}
	}

}

File: versioned_docs/version-1.20.x/examples/py/src/data_ingestion_batch.py
import asyncio

from reduct import Client, Bucket, RecordBatch


async def main():
    # Create a client instance, then get or create a bucket
    async with Client("http://127.0.0.1:8383", api_token="my-token") as client:
        bucket: Bucket = await client.create_bucket("my-bucket", exist_ok=True)

        # Prepare a batch of records for different entries
        batch = RecordBatch()
        batch.add(
            "sensor-1",
            "2024-02-02T10:00:00",
            b"Temperature: 20.5",
            content_type="text/plain",
            labels={"unit": "C"},
        )
        batch.add(
            "sensor-1",
            "2024-02-02T10:00:01",
            b"Temperature: 20.6",
            content_type="text/plain",
            labels={"unit": "C"},
        )
        batch.add(
            "camera-1",
            "2024-02-02T10:00:02",
            b"Frame #1",
            content_type="text/plain",
        )

        # Write records from multiple entries in one request
        errors = await bucket.write_record_batch(batch)

        # Check statuses and raise first error
        for entry_errors in errors.values():
            for _timestamp, err in entry_errors.items():
                raise err


loop = asyncio.get_event_loop()
loop.run_until_complete(main())
